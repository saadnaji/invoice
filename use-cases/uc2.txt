add_type("chocolate")
add_type("ChocoLate")
add_type("water") 
add_product("pepsi", 100)
add_product("water", 1005)
add_product("chocolate", 191000000)
add_product("chocolate", -5)
add_order(<<["nuts", 29], ["bolts", 31], ["pepsi", 300]>>)
add_order(<<["waTer", 22], ["chOcoLate", 5], ["chOcoLate", 5]>>)
add_order(<<["waTer", 22], ["chOcoLate", -5]>>)
add_type("chips")
add_product("chips", 3000)
add_order(<<["chips", 100]>>)
invoice(1)
invoice(1)
cancel_order(10)
invoice(10)
add_type("")
add_order(<<["chips", 100000000]>>)

