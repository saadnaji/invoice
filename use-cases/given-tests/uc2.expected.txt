  report:      ok
  id:          0
  products:   
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("nuts")
  report:      ok
  id:          0
  products:    nuts
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("bolts")
  report:      ok
  id:          0
  products:    bolts,nuts
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("hammers")
  report:      ok
  id:          0
  products:    bolts,hammers,nuts
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("nuts")
  report:      product type already in database
  id:          0
  products:    bolts,hammers,nuts
  stock:       
  orders:      
  carts:       
  order_state: 
->add_product("bolts",100)
  report:      ok
  id:          0
  products:    bolts,hammers,nuts
  stock:       bolts->100
  orders:      
  carts:       
  order_state: 
->add_product("bolts",0)
  report:      quantity added must be positive
  id:          0
  products:    bolts,hammers,nuts
  stock:       bolts->100
  orders:      
  carts:       
  order_state: 
->add_product("bolts",-1)
  report:      quantity added must be positive
  id:          0
  products:    bolts,hammers,nuts
  stock:       bolts->100
  orders:      
  carts:       
  order_state: 
->add_product("junk",200)
  report:      product not in database
  id:          0
  products:    bolts,hammers,nuts
  stock:       bolts->100
  orders:      
  carts:       
  order_state: 
->add_product("nuts",1005)
  report:      ok
  id:          0
  products:    bolts,hammers,nuts
  stock:       bolts->100,nuts->1005
  orders:      
  carts:       
  order_state: 
->add_order(<<["nuts", 1005], ["bolts", 100]>>)
  report:      ok
  id:          1
  products:    bolts,hammers,nuts
  stock:       
  orders:      1
  carts:       1: bolts->100,nuts->1005
  order_state: 1->pending
->add_order(<<["hammers", 1]>>)
  report:      not enough in stock
  id:          1
  products:    bolts,hammers,nuts
  stock:       
  orders:      1
  carts:       1: bolts->100,nuts->1005
  order_state: 1->pending
->add_order(<<["nuts", 1]>>)
  report:      not enough in stock
  id:          1
  products:    bolts,hammers,nuts
  stock:       
  orders:      1
  carts:       1: bolts->100,nuts->1005
  order_state: 1->pending
->invoice(1)
  report:      ok
  id:          1
  products:    bolts,hammers,nuts
  stock:       
  orders:      1
  carts:       1: bolts->100,nuts->1005
  order_state: 1->invoiced
->invoice(2)
  report:      order id is not valid
  id:          1
  products:    bolts,hammers,nuts
  stock:       
  orders:      1
  carts:       1: bolts->100,nuts->1005
  order_state: 1->invoiced
->invoice(1)
  report:      order already invoiced
  id:          1
  products:    bolts,hammers,nuts
  stock:       
  orders:      1
  carts:       1: bolts->100,nuts->1005
  order_state: 1->invoiced
->cancel_order(1)
  report:      ok
  id:          1
  products:    bolts,hammers,nuts
  stock:       bolts->100,nuts->1005
  orders:      
  carts:       
  order_state: 
->add_product("hammers",100)
  report:      ok
  id:          1
  products:    bolts,hammers,nuts
  stock:       bolts->100,hammers->100,nuts->1005
  orders:      
  carts:       
  order_state: 
->add_order(<<["hammers", 2], ["nuts", 10], ["hammers", 1]>>)
  report:      duplicate products in order array
  id:          1
  products:    bolts,hammers,nuts
  stock:       bolts->100,hammers->100,nuts->1005
  orders:      
  carts:       
  order_state: 
->add_order(<<["hammers", 2], ["nuts", 10], ["bolt", 5]>>)
  report:      some products in order not valid
  id:          1
  products:    bolts,hammers,nuts
  stock:       bolts->100,hammers->100,nuts->1005
  orders:      
  carts:       
  order_state: 
